cli/src/main/scala/com/codacy/analysis/cli/analysis/AnalyseExecutor.scala

```diff
@@ -62,7 +62,12 @@ class AnalyseExecutor(formatter: Formatter,
                 .map(dockerTool => new DockerToolDocumentation(dockerTool, new CacheDockerHelper()))
               val toolHasConfigFiles = fileCollector.hasConfigurationFiles(tool, allFiles)
               val analysisResults =
-                issues(tool, filteredFiles, configuration.toolConfiguration, toolHasConfigFiles)
+                issues(
+                  tool,
+                  filteredFiles,
+                  configuration.toolConfiguration,
+                  toolHasConfigFiles,
+                  configuration.tmpDirectory.map(_.toJava))
```

Comment:
Any particular reason to do .toJava every time instead of passing the file you have?

Reply:
The tool, metrics and duplication runner (present on Codacy-plugins) receive a `java.io.File` as parameter.
However, instead of doing this conversion at the AnalyseExecutor level, moved it to the tool level (just before entering Codacy-plugins). This way, we only use this type file on the CLI.

Reply:
Alright 👍